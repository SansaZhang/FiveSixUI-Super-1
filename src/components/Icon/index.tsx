import React from 'react';
import classNames from 'classnames';
import '../style/index.less';

export interface IconProps {
  type: string;
  className?: string;
  title?: string;
  onClick?: React.MouseEventHandler<any>;
  spin?: boolean;
  style?: React.CSSProperties;
}

const Icon = (props: IconProps) => {
  const { type, className = '', spin } = props;
  const classString = classNames(className,{
    anticon: true,
    'anticon-spin': !!spin || type === 'loading',
    [`anticon-${type}`]: true,
  });
  const effectProps = {...props};
  delete effectProps.type;
  delete effectProps.spin;
  delete effectProps.className;
  return <i {...effectProps} className={classString} />;
};

export default Icon;
