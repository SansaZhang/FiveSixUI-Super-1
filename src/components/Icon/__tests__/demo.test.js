import React from 'react';
import Icon from '..';
import renderer from 'react-test-renderer';

test('Tag demo', () => {
  const component = renderer.create(
    <Icon type="cross"/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});