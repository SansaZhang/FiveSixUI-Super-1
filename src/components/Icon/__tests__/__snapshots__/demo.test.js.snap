exports[`test Tag demo 1`] = `
<i
  className="anticon anticon-cross" />
`;
